gnosis/safe-react
# (Fix) Refactor Safe information retrieval

Aims to close #2048 

We're are having this issue where the Apps or Collectibles sections disappear.

One of the possible reasons is that we're managing the update of Safe's information from many sources. Two slightly different functions for the same goal (update safe's info), as well as loading data from the `transaction-service` and from RPC.

Here a refactor was made for the `fetchSafe` action. `checkAndUpdateSafe` action was deleted. And `buildSafe` was refactored too. Now `fetchSafe` and `buildSafe` ~share quite similar logic~ are clearly differentiated.

On the technical side, instead of using `Promise.all`, we used `.allSettled` as it allows for better flexibility while we're retrieving Safe's data. Data is requested from `client-gateway` instead of a mix of `transaction-service` and `RPC calls`.

This refactor aims to leave the functionality untouched (`fetchSafe` and `buildSafe` both do the exact same tasks they were doing until now).

Things I noticed that we should work on from here:
1. load safe flow, can be improved
2. prevent unnecessary calls to `fetchSafe` (especially when loading from URL)
3. continue deleting legacy code.

## How to test
Given that this is a refactor on how we recover and load the safe information, the test should be as thorough as possible.

What I certainly recommend checking is:
* Every 'load safe' flow, should behave as in production.
* Every 'create safe' flow, should behave as in production (with a slight enhancement, see: https://github.com/gnosis/safe-react/pull/2122#issuecomment-821836310)
* No more 'app' or 'collectibles' sections disappearing
